### PR TITLE
Select items

### PR DESCRIPTION
**Pull request by vlad0**
Added rating feature for every item from the product list.

### CODE REVIEW

File: src/client/app/review/items-review/items-review.component.ts [hunk 1]
```diff
@@ -1,8 +1,42 @@
...
+  }
+
+  // public methods
+  rate(listItem:ProductListItem):void {
+    this.reviewService.rateItemFromItemsReview(listItem);
+  }
+
+  removeItem(listItem: ProductListItem):void {
+    // set the rating to 0 so the Review Component
+    // can be updated
```
Comment (ivopashov) [inline src/client/app/review/items-review/items-review.component.ts]: no need of the comments

File: src/client/app/review/product-items/product-items.component.ts [hunk 2]
```diff
@@ -1,6 +1,8 @@
-import {Component, Input, OnChanges, SimpleChange, OnInit} from '@angular/core'
-import {ProductList, ProductCategory} from "../../shared/models/index";
+import {Component, Input, Output, OnChanges, OnInit, EventEmitter, SimpleChange} from '@angular/core'
```
Comment (ivopashov) [inline src/client/app/review/product-items/product-items.component.ts]: I don't see you using many of those imports - why importing them ?

File: src/client/app/review/review.component.ts [hunk 3]
```diff
@@ -1,29 +1,35 @@
 import {Component, OnInit} from '@angular/core';
-import {ProductListService} from "../shared/product-list/index";
+import {ProductListService, ProductListItem} from "../shared/product-list/index";
```
Comment (ivopashov) [inline src/client/app/review/review.component.ts]: you don't use this `ProductListItem` - why need?

**Comment by ivopashov** [comment]
🔨 but please address my  feedback - no need of comments and beware if unused imports.

File: src/client/app/review/product-items/product-items.component.ts [hunk 4]
```diff
@@ -10,9 +12,31 @@ import {FormControl} from '@angular/forms';
...
   private _categories:ProductCategory[];
 
+  // public methods
+  rate(listItem:ProductListItem):void {
+    this.reviewService.rateItemFromProductList(listItem);
+  }
+
+  constructor(private reviewService: ReviewService) {
+    var self = this;
+    this.subscription = this.reviewService.itemsFromItemsReview$.subscribe(value => {
```
Comment (Elinos) [inline src/client/app/review/product-items/product-items.component.ts]: "itemsFromItemsReview$" what does the $ means here?

Comment (vlad0) [inline src/client/app/review/product-items/product-items.component.ts]: convention

File: src/client/app/review/items-review/items-review.component.ts (hunk 1 above)
Comment (vlad0) [inline src/client/app/review/items-review/items-review.component.ts]: they are for personal use.

File: src/client/app/review/product-items/product-items.component.ts (hunk 2 above)
Comment (vlad0) [inline src/client/app/review/product-items/product-items.component.ts]: Thanks!

**Pull request merged**